docs/src/main/tut/chisel3/modules.md

```diff
@@ -84,3 +84,68 @@ children modules, using the ```Module``` constructor function and
 the Scala ```new``` keyword to create a
 new object.  We then wire them up to one another and to the ports of
 the ```Mux4``` interface.
+
+### RawModule
+
+*Module* are used to connect components inside a design. But once we
+want to synthesize for the real world (FPGA, ASIC, ...) some extra
+features are required like io specific naming, reset polarity, clock
+assignement, ...
+
+For this, a specific module named *RawModule* should be used. With RawModule
+there is no implicit signals. All signals like clock and reset should be
+given in IO argument.
+
+Then we can use it in place of *Module* usage :
+```tut:silent
+class SlaveSpi extends Module {
+  val io = IO(new Bundle{
+    val mosi = Input(Bool())
+    val miso = Output(Bool())
+    val sclk = Input(Bool())
+    val csn = Input(Bool())
+  })
+[...]
```

Comment:
Ok, I'm deleting tut shed replacing it with ```scala

Reply:
Actually, if you can keep the `tut` sheds I think that would be better. There's then a guarantee that a user can copy-paste that code block and it will work (with a specific version of the scala compiler).

Going forward, I view `tut` or `mdoc` as the approach for keeping the documentation in sync (and guaranteed to work): https://github.com/freechipsproject/www.chisel-lang.org/issues/4

Reply:
Ok